docs/concepts/did.md

```diff
@@ -2,3 +2,82 @@
+While transaction fees cannot be refunded, the deposit is returned when the DID is deleted from the blockchain: this is to incentivize users to clean the data from the blockchain once such data is not needed anymore.
+
+An example of a full DID is the following:
+
+```
+did:kilt:4rp4rcDHP71YrBNvDhcH5iRoM3YzVoQVnCZvQPwPom9bjo2e
+```
+
+Here, there is no `light:` component, which indicates that the DID is a full DID and that the keys associated with it must not be derived from the DID identifier but must be retrieved from the KILT blockchain.
+
+Beyond an authentication key, an encryption key, and service endpoints, a full DID also supports an **attestation key**, which must be used to write CTypes and attestations on the blockchain, and a **delegation key**, which must be used to write delegations on the blockchain.
+
+## Migrating a light DID to a full DID
+
+The **migration** of a DID means that a light, off-chain DID is anchored to the KILT blockchain, supporting all the features that full DIDs provide.
+In the current version (v1) of the KILT DID protocol, a light DID of the form `did:kilt:light:014nv4phaKc4EcwENdRERuMF79ZSSB5xvnAk3zNySSbVbXhSwS` would become a full DID of the form `did:kilt:4nv4phaKc4EcwENdRERuMF79ZSSB5xvnAk3zNySSbVbXhSwS`.
+Note that the identifier of the two DIDs, apart from the initial `01` sequence of the light DID, are equal since both DIDs are derived from the same KILT account.
+
+Once a light DID is migrated, all the credentials collected by the light DID can only be presented using the migrated on-chain DID.
+This is by design, as it is assumed that the user had valid reasons to migrate the DID on the chain, and as on-chain DIDs offer greater security guarantees, KILT will reject light DID signatures even in case the original claim in the attestation was generated with that off-chain DID.
+
+## DIDs and Web3 names
+
+Web3 names are user-friendly aliases for KILT DIDs.
+They serve the same purpose that domain names do for IP addresses: who knows the IP address under the `kilt.io` domain name? 🤷🏽‍♀️
+There is a one-to-one relationship between DIDs and Web3 names.
+This means that a KILT DID can be linked to at most one Web3 name, and a Web3 name can be claimed only by one DID.
+
+Each Web3 name is globally unique within the KILT space, and is composed of a sequence of a minimum of 3 to a maximum of 32 characters taken from a specific character set to enhance human redability and reduce the chances of "two Web3 names looking the same, despite being different".
+The character set include only:
+- *lowercase letters*, from `a` to `z`
+- *digits* from `0` to `9`
+- the symbols `-` and `_`
+
+A regex that would match all and only the allowed Web3 names would be the following:
+
+```
+^[a-z0-9\_\-]{3,32}$
+```
+
+In the global URI space, Web3 names are prefixed with the `w3n:` URI namespace. 
+For example, the full URI for the Web3 name `example-web3name` is `w3n:example-web3name`.
+
+### Linking accounts to a Web3 name
```

Comment:
I would love to see a simple diagram for this. Just to make it easier.

Reply:
Definitely! We need to step up our diagram game. If you want to, I can provide these @ntn-x2 .

Reply:
Phew! This cost me quite some time! Let me know what you guys think https://github.com/KILTprotocol/docs/pull/82/commits/f2ebb7497232d9fc2c92a65344f9722c4f7908ca.